src/experimental/constructs/iam/roles/cross-account-role.test.ts

```diff
@@ -0,0 +1,76 @@
+import { Template } from "aws-cdk-lib/assertions";
+import { Effect, PolicyStatement, ServicePrincipal } from "aws-cdk-lib/aws-iam";
+import { GuRole } from "../../../../constructs/iam";
+import { simpleGuStackForTesting } from "../../../../utils/test";
+import { GuCrossAccountRoleExperimental } from "./cross-account-role";
+
+describe("The GuCrossAccountRoleExperimental construct", () => {
+  it("can create a cross account role", () => {
+    const stack = simpleGuStackForTesting();
+    new GuCrossAccountRoleExperimental(stack, "testCrossAccountRole", {
+      nameOfRoleWhichCanAssumeThisRole: "nameOfRoleInOtherAccountWhichCanAssumeThis-STAGE",
+      roleName: "crossAccountRole",
+      accountId: "1234",
+    });
+    expect(Template.fromStack(stack).toJSON()).toMatchSnapshot();
+  });
+
+  it("can create a cross account role that can be assumed by a service in another account", () => {
+    const stackInAccountB = simpleGuStackForTesting();
+    const crossAccountRole = new GuCrossAccountRoleExperimental(stackInAccountB, "testCrossAccountRole", {
+      nameOfRoleWhichCanAssumeThisRole: "roleInAccountA-CODE",
+      roleName: "roleInAccountB-CODE",
+      accountId: "idForAccountA",
+    });
+
+    // This is just an example: after creating the cross account role we can add policies which allow the role in
+    // the other account to perform the given actions in this account.
+    crossAccountRole.addToPolicy(
+      new PolicyStatement({
+        actions: ["sqs:SendMessage"],
+        resources: ["queue.queueArn"],
+      }),
+    );
+
+    Template.fromStack(stackInAccountB).hasResourceProperties("AWS::IAM::Role", {
+      RoleName: "roleInAccountB-CODE",
+      AssumeRolePolicyDocument: {
+        Statement: [
+          {
+            Action: "sts:AssumeRole",
+            Effect: "Allow",
+            Principal: {
+              AWS: "arn:aws:iam::idForAccountA:role/roleInAccountA-CODE",
+            },
+          },
+        ],
+      },
+    });
+
+    const stackInAccountA = simpleGuStackForTesting();
+    const role = new GuRole(stackInAccountA, "idForRole", {
+      assumedBy: new ServicePrincipal("ec2.amazonaws.com"),
```

Comment:
Minor: as we're adding a test for documentation purposes, it could be worth a comment here to explain that (IIUC!) this corresponds to the [AWS service](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements_principal.html#principal-services) that our app will be running in. 

In other words, `ec2.amazonaws.com` is just an example and it could be something else depending on the user's runtime e.g. `lambda.amazonaws.com`.